src/pgen/torus.cpp

```diff
@@ -532,18 +545,178 @@ void PostInitializationModifier(ParameterInput *pin, Mesh *pmesh) {
     printf("Torus normalization factor = %.14e\n", B_field_fac);
   }
 
+  // reduction: integrate over whole disk for tracers number
+  // Need this global quantity before setting up tracers
+  Real number_mesh = 0.0;
+  for (auto &pmb : pmesh->block_list) {
+    auto &rc = pmb->meshblock_data.Get();
+    auto tracer_pkg = pmb->packages.Get("tracers");
+    bool do_tracers = tracer_pkg->Param<bool>("active");
+
+    if (do_tracers) {
+      auto geom = Geometry::GetCoordinateSystem(rc.get());
+      auto coords = pmb->coords;
+
+      // set up transformation stuff
+      auto gpkg = pmb->packages.Get("geometry");
+      bool derefine_poles = gpkg->Param<bool>("derefine_poles");
+      Real h = gpkg->Param<Real>("h");
+      Real xt = gpkg->Param<Real>("xt");
+      Real alpha = gpkg->Param<Real>("alpha");
+      Real x0 = gpkg->Param<Real>("x0");
+      Real smooth = gpkg->Param<Real>("smooth");
+      auto tr = Geometry::McKinneyGammieRyan(derefine_poles, h, xt, alpha, x0, smooth);
+
+      auto ib = pmb->cellbounds.GetBoundsI(IndexDomain::interior);
+      auto jb = pmb->cellbounds.GetBoundsJ(IndexDomain::interior);
+      auto kb = pmb->cellbounds.GetBoundsK(IndexDomain::interior);
+
+      /* integrate f_atmosphere dV on blocks, with f_atmosphere = {0,1}, and MPI reduce */
+      pmb->par_reduce(
+          "Phoebus::ProblemGenerator::Torus::MeshTracerNumber", kb.s, kb.e, jb.s, jb.e,
+          ib.s, ib.e,
+          KOKKOS_LAMBDA(const int k, const int j, const int i, Real &disk_vol) {
+            const Real x1 = coords.Xc<1>(k, j, i);
+            const Real x2 = coords.Xc<2>(k, j, i);
+            Real r = tr.bl_radius(x1);
+            Real th = tr.bl_theta(x1, x2);
+
+            Real lnh = -1.0;
+            Real uphi;
+            if (r > rin) lnh = log_enthalpy(r, th, a, rin, angular_mom, uphi);
+            if (lnh > 0.0) {
+              const Real vol = coords.CellVolume(k, j, i);
```

Comment:
This should have a factor of the metric in it I think. (It probably doesn't matter for problems we run, but we should include it.)